diff -Nur stackless-2.6.2-orig//configure stackless-2.6.2/configure
--- stackless-2.6.2-orig//configure	2010-05-20 17:33:32.999079687 +0200
+++ stackless-2.6.2/configure	2010-05-20 17:43:58.138186423 +0200
@@ -2648,7 +2648,7 @@
 See \`config.log' for more details." >&5
 echo "$as_me: error: no acceptable C compiler found in \$PATH
 See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+}
 
 # Provide some information about the compiler.
 echo "$as_me:$LINENO: checking for C compiler version" >&5
Binary files stackless-2.6.2-orig//.configure.swp and stackless-2.6.2/.configure.swp differ
diff -Nur stackless-2.6.2-orig//Makefile.pre.in stackless-2.6.2/Makefile.pre.in
--- stackless-2.6.2-orig//Makefile.pre.in	2010-05-20 17:33:32.989078907 +0200
+++ stackless-2.6.2/Makefile.pre.in	2010-05-20 17:37:52.429050922 +0200
@@ -182,6 +182,9 @@
 
 PYTHON=		python$(EXE)
 BUILDPYTHON=	python$(BUILDEXE)
+HOSTPYTHON=	python2.6
+
+CROSS_COMPILE=	yes
 
 # The task to run while instrument when building the profile-opt target
 PROFILE_TASK=	$(srcdir)/Tools/pybench/pybench.py -n 2 --with-gc --with-syscheck
@@ -411,8 +414,8 @@
 # Build the shared modules
 sharedmods: $(BUILDPYTHON)
 	@case $$MAKEFLAGS in \
-	*s*) $(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' ./$(BUILDPYTHON) -E $(srcdir)/setup.py -q build;; \
-	*) $(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' ./$(BUILDPYTHON) -E $(srcdir)/setup.py build;; \
+	*s*) $(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' $(BUILDPYTHON) -E $(srcdir)/setup.py -q build;; \
+	*) $(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' $(BUILDPYTHON) -E $(srcdir)/setup.py build;; \
 	esac
 
 # Build static library
Binary files stackless-2.6.2-orig//.Makefile.pre.in.swp and stackless-2.6.2/.Makefile.pre.in.swp differ
